Author: Luca Boccassi <luca.boccassi@gmail.com>
Description: Fix kernel module build on 4.10 and greater
 From kernel 4.10 and newer (commit 1a29d85eb0f19) virtual_address is no longer
 available as a member of struct vm_fault. Given it's in a header it causes a
 build failure in all modules, but upstream's conftest.sh only does the check
 when building the nvidia-uvm module. Add the check for the main module too.
--- a/nvidia/nvidia.Kbuild
+++ b/nvidia/nvidia.Kbuild
@@ -162,6 +162,7 @@
 NV_CONFTEST_TYPE_COMPILE_TESTS += dma_map_ops
 NV_CONFTEST_TYPE_COMPILE_TESTS += noncoherent_swiotlb_dma_ops
 NV_CONFTEST_TYPE_COMPILE_TESTS += vm_fault_present
+NV_CONFTEST_TYPE_COMPILE_TESTS += vm_fault_has_address
 
 NV_CONFTEST_GENERIC_COMPILE_TESTS += dom0_kernel_present
 NV_CONFTEST_GENERIC_COMPILE_TESTS += nvidia_vgpu_kvm_build
